src/hotspot/cpu/x86/stubGenerator_x86_64.cpp

```diff
@@ -4179,6 +4179,57 @@ void StubGenerator::generate_compiler_stubs() {
       = CAST_FROM_FN_PTR(address, SharedRuntime::montgomery_square);
   }
 
+  // Load x86_64_sort library on supported hardware to enable avx512 sort and partition intrinsics
+    if (UseAVX > 2 && VM_Version::supports_avx512dq()) {
+
+      void *libsimdsort = nullptr;
+      char ebuf_[1024];
+      char dll_name_simd_sort[JVM_MAXPATHLEN];
+      if (os::dll_locate_lib(dll_name_simd_sort, sizeof(dll_name_simd_sort), Arguments::get_dll_dir(), "simdsort")) {
+        libsimdsort = os::dll_load(dll_name_simd_sort, ebuf_, sizeof ebuf_);
+      }
+    // Get addresses for avx512 sort and partition routines
+    if (libsimdsort != nullptr) {
+      log_info(library)("Loaded library %s, handle " INTPTR_FORMAT, JNI_LIB_PREFIX "x86_64" JNI_LIB_SUFFIX, p2i(libsimdsort));
+
+      snprintf(ebuf_, sizeof(ebuf_), "avx512_sort_int");
+      StubRoutines::_arraysort_int = (address)os::dll_lookup(libsimdsort, ebuf_);
+
+      snprintf(ebuf_, sizeof(ebuf_), "avx512_sort_long");
+      StubRoutines::_arraysort_long = (address)os::dll_lookup(libsimdsort, ebuf_);
+
+      snprintf(ebuf_, sizeof(ebuf_), "avx512_sort_float");
+      StubRoutines::_arraysort_float = (address)os::dll_lookup(libsimdsort, ebuf_);
+
+      snprintf(ebuf_, sizeof(ebuf_), "avx512_sort_double");
+      StubRoutines::_arraysort_double = (address)os::dll_lookup(libsimdsort, ebuf_);
```

Comment:
Hi @vamsi-parasa , If we align these C++ stub entry points with java intrinsic entry points which accept an element class argument, it will significantly cleanup the code referring to these stub entry points. We can pass an extra int type flag to stubs based on which C++ implementation can appropriately call type specific leaf level routines.

Reply:
Hello Jatin, the suggested refactoring change is incorporated in the latest commit.